### PR TITLE
Add support for runner_on_start

### PR DESCRIPTION
**Pull request by matburt**
This will be available in ansible 2.8
implements: https://github.com/ansible/awx/issues/2539

### CODE REVIEW

**Comment by AlanCoding** [comment]
I checked out this branch and ran a debug playbook. I don't see any runner_on_start events.

I suspect this is because no stdout is produced during this task.

**Comment by AlanCoding** [comment]
oh wait, need Ansible devel install

**Comment by AlanCoding** [comment]
I updated to devel and I still don't see the event, although it seems to streaming standard out might have been messed up. I'll have to come back to retest later.

**Comment by AlanCoding** [comment]
hmm... these logs just popped up

```
awx_1        | 18:48:20 dispatcher.1 | 2018-10-30 18:48:20,597 INFO     awx.main.queue.CallbackQueueDispatcher Publish Job Event Exception: EncodeError(TypeError('Object of type Host is not JSON serializable',),), retry=3
awx_1        | 18:48:20 dispatcher.1 | Traceback (most recent call last):
awx_1        | 18:48:20 dispatcher.1 |   File "/awx_devel/awx/main/queue.py", line 49, in dispatch
awx_1        | 18:48:20 dispatcher.1 |     routing_key=self.connection_queue)
awx_1        | 18:48:20 dispatcher.1 |   File "/venv/awx/lib/python2.7/site-packages/kombu/messaging.py", line 169, in publish
awx_1        | 18:48:20 dispatcher.1 |     compression, headers)
awx_1        | 18:48:20 dispatcher.1 |   File "/venv/awx/lib/python2.7/site-packages/kombu/messaging.py", line 252, in _prepare
awx_1        | 18:48:20 dispatcher.1 |     body) = dumps(body, serializer=serializer)
awx_1        | 18:48:20 dispatcher.1 |   File "/venv/awx/lib/python2.7/site-packages/kombu/serialization.py", line 221, in dumps
awx_1        | 18:48:20 dispatcher.1 |     payload = encoder(data)
awx_1        | 18:48:20 dispatcher.1 |   File "/usr/lib64/python2.7/contextlib.py", line 35, in __exit__
awx_1        | 18:48:20 dispatcher.1 |     self.gen.throw(type, value, traceback)
awx_1        | 18:48:20 dispatcher.1 |   File "/venv/awx/lib/python2.7/site-packages/kombu/serialization.py", line 54, in _reraise_errors
awx_1        | 18:48:20 dispatcher.1 |     reraise(wrapper, wrapper(exc), sys.exc_info()[2])
awx_1        | 18:48:20 dispatcher.1 |   File "/venv/awx/lib/python2.7/site-packages/kombu/serialization.py", line 50, in _reraise_errors
awx_1        | 18:48:20 dispatcher.1 |     yield
awx_1        | 18:48:20 dispatcher.1 |   File "/venv/awx/lib/python2.7/site-packages/kombu/serialization.py", line 221, in dumps
awx_1        | 18:48:20 dispatcher.1 |     payload = encoder(data)
awx_1        | 18:48:20 dispatcher.1 |   File "/venv/awx/lib/python2.7/site-packages/kombu/utils/json.py", line 69, in dumps
awx_1        | 18:48:20 dispatcher.1 |     **dict(default_kwargs, **kwargs))
awx_1        | 18:48:20 dispatcher.1 |   File "/venv/awx/lib/python2.7/site-packages/simplejson/__init__.py", line 399, in dumps
awx_1        | 18:48:20 dispatcher.1 |     **kw).encode(obj)
awx_1        | 18:48:20 dispatcher.1 |   File "/venv/awx/lib/python2.7/site-packages/simplejson/encoder.py", line 291, in encode
awx_1        | 18:48:20 dispatcher.1 |     chunks = self.iterencode(o, _one_shot=True)
awx_1        | 18:48:20 dispatcher.1 |   File "/venv/awx/lib/python2.7/site-packages/simplejson/encoder.py", line 373, in iterencode
awx_1        | 18:48:20 dispatcher.1 |     return _iterencode(o, 0)
awx_1        | 18:48:20 dispatcher.1 |   File "/venv/awx/lib/python2.7/site-packages/kombu/utils/json.py", line 59, in default
awx_1        | 18:48:20 dispatcher.1 |     return super(JSONEncoder, self).default(o)
awx_1        | 18:48:20 dispatcher.1 |   File "/venv/awx/lib/python2.7/site-packages/simplejson/encoder.py", line 268, in default
awx_1        | 18:48:20 dispatcher.1 |     o.__class__.__name__)
awx_1        | 18:48:20 dispatcher.1 | EncodeError: Object of type Host is not JSON serializable
```

File: awx/lib/awx_display_callback/module.py [hunk 1]
```diff
@@ -475,6 +475,15 @@ def v2_runner_retry(self, result):
         with self.capture_event_data('runner_retry', **event_data):
             super(BaseCallbackModule, self).v2_runner_retry(result)
 
+    def v2_runner_on_start(self, host, task):
+        event_data = dict(
+            host=host,
```
Comment (ryanpetrello) [inline awx/lib/awx_display_callback/module.py]: Seems like this isn't serializiable, given Alan's comment.  Should we `str()` this or is there a `.name` property or something?

**Review by ryanpetrello — CHANGES_REQUESTED** [review]
Comments inline, I'd prefer an integration test for this one before merging.

File: awx/lib/awx_display_callback/module.py (hunk 1 above)
Comment (matburt) [reply awx/lib/awx_display_callback/module.py]: I'll check up on it and see wtf it is.

**Comment by matburt** [comment]
I think this actually makes for a better integration test scenario on runner rather than on AWX. I'll add one over there, we also have a bit more flexibility on testing event content on the runner side.

**Comment by matburt** [comment]
Tests on the runner side here: https://github.com/ansible/ansible-runner/pull/165

**Review by ryanpetrello — APPROVED** [review]

**Comment by AlanCoding** [comment]
looks good:

https://gist.github.com/AlanCoding/1739c6f639cb2737046eff6a455ebc29

**Review by AlanCoding — APPROVED** [review]

**Pull request merged**